### ISSUE TITLE
ssss

### ISSUE THREAD

**Issue by sena1989** [sena1989/angularjslogin1#2]
html.

<!DOCTYPE html>
<html>
<head>
    <meta charset="ISO-8859-1">
    <title>Main Page</title>
    <link rel="stylesheet" href="css/lib/bootstrap.min.css"/>
     <link rel="stylesheet" href="css/app/app.css"/>
  <link href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.6-rc.0/css/select2.min.css" rel="stylesheet" />
 <script src="js/lib/angular.min.js"></script>
<script src="js/lib/ui-bootstrap-tpls-2.5.0.min.js"></script>
  <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.6-rc.0/js/select2.min.js"></script>
<script src="js/lib/ui-bootstrap-tpls-2.5.0.min.js"></script> 
<script src="js/lib/bootstrap.min.js"></script>
<script src="js/app/app.js"></script>
<script src="js/app/Controller.js"></script>
<script src="js/app/Service.js"></script>
<script src="js/lib/angular-route.js"></script>
 <script>
  $(document).ready(function() {
    $('.js-example-basic-multiple').select2();
});
    </script>
</head>
<body ng-app="DBAutomation">
<h2>STAR HISTORY ENCRYPTION</h2>
<hr/>
<form ng-submit="formSubmit()" class="form">
<div class="container" ng-controller="DBAutomationController">
     
    <div class="row">
  <div class="col-md-6 ">
    <label style="padding-left: 274px;" for="uname"><b>Table to be encrypted :</b></label>
	</div>	
    <div class="col-md-6  custom-select" style="width:200px;">
   <select ng-model="myselect" ng-options="o as o for o in allTables" style="width: 135%;" class="input-sm spacing" ng-change="getTableColumn(myselect)">
  </select>
</div>
</div>
<br>
<div class="row">
  
  <div class="col-md-6 ">
    <label style="padding-left: 274px;" for="uname"><b>Columns to be encrypted :</b></label>
	</div>
	
    <div class="col-md-6  custom-select" style="width: 50%;">
   
 <label ng-repeat="columnName in selectedTableColumn"> <input
				type="checkbox" name="selectedEncrypt[]" value="{{columnName}}"
				ng-checked="encryptSelection.indexOf(columnName) > -1"
				ng-click="selectEncryptColumn(columnName)">
				{{columnName}}
			</label> {{encryptSelection}} 
			 
 
</div>
</div>
<br>
<div class="row">
  
  <div class="col-md-6 ">
    <label style="padding-left: 274px;" for="uname"><b>Batch Column :</b></label>
	</div>
	
    <div class="col-md-6  custom-select" style="width:50%;">
  <label ng-repeat="columnName in selectedTableColumn"> <input
				type="checkbox" name="selectedBatch[]" value="{{columnName}}"
				ng-checked="batchSelection.indexOf(columnName) > -1"
				ng-click="selectBatchColumn(columnName)">
				{{columnName}}
			</label> {{batchSelection}}
 
</div>
</div>
<br>
<div class="row">
  
  <div class="col-md-6 ">
    <label style="padding-left: 274px;" style="padding-left: 274px;" for="uname"><b>Primary key columns :</</b></label>
       </div>
       
<div class="col-md-6  custom-select" style="width:50%;">
       
<label ng-repeat="columnName in selectedTableColumn"> <input
					type="checkbox" name="selectedPrimary[]" value="{{columnName}}"
					ng-checked="primarySelection.indexOf(columnName) > -1"
					ng-click="selectPrimaryColumn(columnName)"> {{columnName}}
				</label> {{primarySelection}}

</div>
</div>
<br>

<br>
<div class="row">
<div class="col-md-4 ">
</div>
<div class="col-md-4 " style="width: 39.333333%;">
<button type="submit" ng-click="executeProcedure()">Submit</button>
</div>
<div class="col-md-4 ">
</div>
</div>
  
</div>

<div data-ng-view="" id="ng-view" class="slide-animation"></div>
</form>
</body>

</html>

-------------
controler

'use strict'

var module = angular.module('DBAutomation.controllers', []);
module.controller("DBAutomationController", [
		"$location",
		"$scope",
		"DBAutomationService",
		"$http",
		function($location, $scope, DBAutomationService, $http) {

			$scope.path = $location.$$protocol + "://" + $location.$$host + ":"
					+ $location.$$port;
			$location.$$port;

			$scope.selectedTable = "";
			$scope.primaryCoulmn = [];

			$scope.allTables = [];
			$scope.selectedTableColumn = [];

			$scope.getEverything = function() {
				$http({
					method : 'GET',
					url : $scope.path + '/getAllTables'
				}).then(function(response) {

					$scope.allTables = response.data;
				}, function(error) {
				});
			};

			$scope.executeProcedure = function() {
				alert("successfully submitted");
			}

			$scope.getEverything();

			$scope.getTableColumn = function(tableName) {
				
				$scope.primarySelection = [];
				$scope.batchSelection = [];
				$scope.encryptSelection = [];

				$http({
					method : 'GET',
					url : $scope.path + '/getTableColumns',
					params : {
						tableName : tableName
					}
				}).then(function(response) {
					$scope.fruits = response.data;
					$scope.selectedTableColumn = response.data;
				}, function(error) {
				});

			}

			$scope.primarySelection = [];
			$scope.batchSelection = [];
			$scope.encryptSelection = [];

			$scope.selectPrimaryColumn = function(columnName) {
				var idx = $scope.primarySelection.indexOf(columnName);

				// Is currently selected
				if (idx > -1) {
					$scope.primarySelection.splice(idx, 1);
				}
				// Is newly selected
				else {
					$scope.primarySelection.push(columnName);
				}
			};
			
			$scope.selectBatchColumn = function(columnName) {
				var idx = $scope.batchSelection.indexOf(columnName);

				// Is currently selected
				if (idx > -1) {
					$scope.batchSelection.splice(idx, 1);
				}
				// Is newly selected
				else {
					$scope.batchSelection.push(columnName);
				}
			};
			
			$scope.selectEncryptColumn = function(columnName) {
				var idx = $scope.encryptSelection.indexOf(columnName);

				// Is currently selected
				if (idx > -1) {
					$scope.encryptSelection.splice(idx, 1);
				}
				// Is newly selected
				else {
					$scope.encryptSelection.push(columnName);
				}
			};
			
			

		} ]);

----------

apps
'use strict'

var app = angular.module('DBAutomation', [  'DBAutomation.controllers',
		'DBAutomation.services', 'ngRoute' ]);
app.constant("CONSTANTS", {
	
	getAllTables : "/getAllTables",
});

app.config(function($routeProvider) {
    $routeProvider
    .when("/mainLayout", {
        templateUrl : '/pages/mainLayout.html',
        controller: 'DBAutomationController'
    })
    /*.when("/dbAutomation", {
        templateUrl : '/static/views/index.html',
        controller: 'DBAutomationController'
    })
    .otherwise(
            { redirectTo: '/'}
        )*/;
    
});

---
'use strict'

angular.module('DBAutomation.services', []).factory('DBAutomationService',
		[ "$http", function($http) {
			var service = {};
			/*service.getUserById = function(userId) {
				var url = CONSTANTS.getUserByIdUrl + userId;
				return $http.get(url);
			}
			service.getAllUsers = function() {
				return $http.get(CONSTANTS.getAllUsers);
			}
			service.saveUser = function(userDto) {
				return $http.post(CONSTANTS.saveUser, userDto);
			}*/
			service.getAllTables = function() {
				return $http.get(getAllTables);
			}
			return service;
		} ]);
--
dbautomationcon

package com.fdc.automation.controller;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

import com.fdc.automation.repository.SampleDAOImpl;
import com.fdc.automation.utils.Constants;
@Controller
public class DBAutomationController {
	@Autowired
	SampleDAOImpl dao;
	
	@GetMapping("/getAllTables")
	@ResponseBody
	public List<String> getAllTables(){
		
		List<String> allTables = null;
		
		
		List<String> tableList = dao.getTableList();
		System.out.println(tableList);
		
		return tableList;
	}
	
	@RequestMapping("/getTableColumns")
    @ResponseBody
    public List<String> getTableColumns(@RequestParam(name="tableName", required=false) String tableName){
           List<String> allTableColumn = null;
           
           List<String> tableColumnList = dao.getTableColumnList(tableName);
           
           return tableColumnList;
    }
	/*@RequestMapping("/executeSP")
    @ResponseBody
    public List<String> executeSP(@RequestParam(name="tableName", required=false) String tableName){
           List<String> allTableColumn = null;
           List<String> executeSPList = dao.setExecuteSPList(tableName);
           return executeSPList;
    }*/

	
	@RequestMapping("/")
	 public String home() {
	  return "index";
	 }
	
	@RequestMapping("/mainLayout")
	public String mainLayout() {
	 return "pages/mainLayout";
	}

}
-----
package com.fdc.automation.repository;

import java.util.List;
import java.util.Map;

import javax.sql.DataSource;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;
import org.springframework.stereotype.Service;

@Configuration
@Repository
public class SampleDAOImpl {
	
	 
	@Autowired
    private DataSource dataSource;
	@Autowired
	private JdbcTemplate jdbcTemplate;
	
	/*public void setDataSource(DataSource dataSource) {
        this.dataSource = dataSource;
        this.jdbcTemplate = new JdbcTemplate(dataSource);
	}*/

	 
	 public List<String> getTableList(){
		 String sql = "select distinct table_name from all_tables where owner !='SYS'";
		 jdbcTemplate.setMaxRows(150);
		 List<String> queryForList = jdbcTemplate.queryForList(sql, String.class);
		 return queryForList;
	 }
	 
	 public List<String> getTableColumnList(String table){
         String sql = "select column_name from all_tab_cols where table_name='"+table+"'";
         jdbcTemplate.setMaxRows(100);
         List<String> queryForList = jdbcTemplate.queryForList(sql, String.class);
         return queryForList;
  }
	 
/*	 public List<String> setExecuteSPList(String table){
         String sql = "select column_name from all_tab_cols where table_name='"+table+"'";
         jdbcTemplate.setMaxRows(100);
         List<String> queryForList = jdbcTemplate.queryForList(sql, String.class);
         return queryForList;
  }*/

}
